web/packages/shared/components/UnifiedResources/types.ts

```diff
@@ -100,3 +104,84 @@ export type UnifiedResourcesQueryParams = {
   // TODO(bl-nero): Remove this once filters are expressed as advanced search.
   kinds?: string[];
 };
+export interface UnifiedResourceViewItem {
+  name: string;
+  labels: {
+    name: string;
+    value: string;
+  }[];
+  primaryIconName: ResourceIconName;
+  SecondaryIcon: typeof Icon;
+  ActionButton: React.ReactElement;
+  cardViewProps: CardViewSpecificProps;
+  listViewProps: ListViewSpecificProps;
+}
+
+export enum PinningSupport {
+  Supported = 'Supported',
+  /**
+   * Disables pinning functionality if a leaf cluster hasn't been upgraded yet.
+   * Shows an appropriate message on hover.
+   * */
+  NotSupported = 'NotSupported',
+  /** Disables the pinning button. */
+  Disabled = 'Disabled',
+  /** Hides the pinning button completely. */
+  Hidden = 'Hidden',
+}
+
+export type ResourceItemProps = {
+  name: string;
+  primaryIconName: ResourceIconName;
+  SecondaryIcon: typeof Icon;
+  cardViewProps?: CardViewSpecificProps;
+  listViewProps?: ListViewSpecificProps;
+  labels: ResourceLabel[];
+  ActionButton: React.ReactElement;
+  onLabelClick?: (label: ResourceLabel) => void;
+  pinResource: () => void;
+  selectResource: () => void;
+  selected: boolean;
+  pinned: boolean;
+  pinningSupport: PinningSupport;
+};
+
+// Props that are needed for the Card view.
+// The reason we need this separately defined is because unlike with the list view, what we display in the
+// description sections of a card varies based on the type of its resource. For example, for applications,
+// instead of showing the `Application` type under the name like we would for other resources, we show the description.
+type CardViewSpecificProps = {
+  primaryDesc?: string;
```

Comment:
`primaryDescription`? In the `ListViewSpecificProps` type we have `description`.

Reply:
We do this because `primaryDescription` can be any arbitrary text we decide it to be and is just defining "the main text under the name." Whereas for the list view, `description` represents the actual `description` field in the resource that's returned from the backend.

Reply:
I named it this because `primaryDescription` can be any arbitrary text we decide it to be and is essentially just describing the main text under the name in the card. This is essentially the same way we named it before (`description.primary`).

Whereas for the list view, `description` directly represents the actual `description` field in the resource that's returned from the backend, and we don't arbitrarily populate it.